linera-execution/src/lib.rs

```diff
@@ -42,22 +36,30 @@ use linera_base::{
     ownership::ChainOwnership,
 };
 use linera_views::{batch::Batch, views::ViewError};
-pub use policy::ResourceControlPolicy;
-pub use resources::{ResourceController, ResourceTracker};
 use serde::{Deserialize, Serialize};
-pub use system::{
-    SystemExecutionError, SystemExecutionStateView, SystemMessage, SystemOperation, SystemQuery,
-    SystemResponse,
-};
 use thiserror::Error;
+
+#[cfg(with_testing)]
+pub use self::applications::ApplicationRegistry;
 #[cfg(all(with_testing, any(with_wasmer, with_wasmtime)))]
-pub use wasm::test as wasm_test;
+pub use self::wasm::test as wasm_test;
 #[cfg(with_wasm_runtime)]
-pub use wasm::{
+pub use self::wasm::{
     ContractEntrypoints, ContractSystemApi, ServiceEntrypoints, ServiceSystemApi, SystemApiData,
     ViewSystemApi, WasmContractModule, WasmExecutionError, WasmServiceModule,
 };
-
+pub use self::{
+    applications::{
+        ApplicationRegistryView, BytecodeLocation, UserApplicationDescription, UserApplicationId,
+    },
+    execution::ExecutionStateView,
+    policy::ResourceControlPolicy,
+    resources::{ResourceController, ResourceTracker},
+    system::{
+        SystemExecutionError, SystemExecutionStateView, SystemMessage, SystemOperation,
+        SystemQuery, SystemResponse,
+    },
+};
 pub use crate::runtime::{ContractSyncRuntime, ServiceSyncRuntime};
```

Comment:
What's the rationale for `self` vs `crate`?

Reply:
Good question, I think it's just a style choice I had on auto-pilot :sweat_smile: I'llchange it to `crate`.